libkbfs/connection_status.go

```diff
@@ -50,7 +50,13 @@ func (kcs *kbfsCurrentStatus) PushConnectionStatusChange(service string, err err
 	defer kcs.lock.Unlock()
 
 	if err != nil {
+		// Exit early if the service is already failed, to avoid an
+		// invalidation.
+		_, errExisted := kcs.failingServices[service]
```

Comment:
To be clear, this is because the previous error would have already invalidated, correct? Have we seen any panics due to the potential double close below?